pkg/api/validate/dynamic/workersku.go

```diff
@@ -0,0 +1,53 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/computeskus"
+)
+
+func (dv *dynamic) ValidateWorkerSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+	// Get a list of available worker SKUs, filtering by location.
+	filter := fmt.Sprintf("location eq %s", location)
+	skus, err := dv.resourceSkusClient.List(ctx, filter)
+	if err != nil {
+		return err
+	}
+
+	sortedSkus := computeskus.FilterVmSizes(skus, location)
+
+	// Validate sku availability for each WorkerProfile
```

Comment:
Would you please provide high level explanation for this loop? I see what it is doing, and I believe I understand why it is happening. It would be easier for everyone to have the explanation lined up here, so they do not have to dig deep and spend hours figuring this one out.

Please think of it as a note for future you, that you would like to see here.